db/revision_cache_lru.go

```diff
@@ -181,6 +185,15 @@ func (rc *LRURevisionCache) Put(docRev DocumentRevision) {
 	value.store(docRev)
 }
 
+// Update a revision
+func (rc *LRURevisionCache) Update(docRev DocumentRevision) {
+	value := rc.getValue(docRev.DocID, docRev.RevID, false)
```

Comment:
As discussed, need to lock for both the get and remove to avoid races.  Can probably simplify and do less work than getValue is doing.